web/src/main/java/com/marklogic/hub/web/service/FlowJobService.java

```diff
@@ -40,17 +40,18 @@
 
     //use a cache with ttl 30 minutes or longer to reduce frequency for fetching data from DB
```

Comment:
The comment is out of sync with the code now - I recommend removing the ttl part so don't have to keep it in sync, just say "use a chance to reduce...".

Reply:
Well, I think I'd like to remove it entirely, as MarkLogic should be a source of truth

Reply:
well, I felt would it be nice to have a cache (5 min is good, 30 min may be too optimistic) in web-tier to avoid unnecessary query against job db (even if ml database has cache mechanism). Unless a user feel the data is not consistent because other apps or he manually changed the data in the database.

Reply:
we might have bug to release client for error handling...